scripts/languageserver/protocol.jl

```diff
@@ -0,0 +1,200 @@
+# Position
+type Position
+    line::Int
+    character::Int
+end
+Position(d::Dict) = Position(d["line"],d["character"])
+Position(line) = Position(line,0)
+Position() = Position(-1,-1)
+isempty(p::Position) = p.line==-1 && p.character==-1
+
+#type Range
+#    start::Position
+#    finish::Position
+#end
+let ex=:(type Range
+        start::Position
+        finish::Position
+    end)
+    ex.args[3].args=ex.args[3].args[[2;4]]
+    ex.args[3].args[2].args[1]=Symbol("end")
+    eval(ex)
+end
```

Comment:
Extremely ugly, I just bodged it together. Do you know of a way to use a keyword as a fieldname?